Sources/PropertyWrappers.swift

```diff
@@ -36,30 +36,46 @@ public struct SwiftyUserDefaultOptions: OptionSet {
 }
 
 @propertyWrapper
-public final class SwiftyUserDefault<T: DefaultsSerializable> where T.T == T {
+public final class SwiftyUserDefault<T: DefaultsSerializable, KeyStore: DefaultsKeyStore> where T.T == T {
 
     public let key: DefaultsKey<T>
     public let options: SwiftyUserDefaultOptions
+    private var customAdapter: DefaultsAdapter<KeyStore>?
```

Comment:
I tried to use single `adapter` property which is custom if provided, otherwise `Defaults`. But didn't workout due to issue with generics.